src/PeliasSearchManager.swift

```diff
@@ -157,33 +157,38 @@ open class PeliasResponse: APIResponse {
   /// The url response if the request completed successfully.
   open let response: URLResponse?
   /// The error if an error occured executing the operation.
-  open let error: NSError?
+  open var error: NSError?
   ///
   open var parsedResponse: PeliasSearchResponse?
-  
+
+  private static let validTypes: Set = ["Point", "MultiPoint", "LineString", "MultiLineString", "Polygon", "MultiPolygon", "GeometryCollection", "Feature", "FeatureCollection"]
+
   public init(data: Data?, response: URLResponse?, error: NSError?) {
     self.data = data
     self.response = response
     self.error = error
-    if let dictResponse = parseData(data) {
-      parsedResponse = PeliasSearchResponse(parsedResponse: dictResponse)
+    if let dictResponse: Dictionary<String, Any> = parseData(data) {
+      if let type = dictResponse["type"] as? String {
+        if PeliasResponse.validTypes.contains(type) {
+          parsedResponse = PeliasSearchResponse(parsedResponse: dictResponse)
+        }
+      } else {
+        let meta = dictResponse["meta"] as? NSDictionary
+        guard let code = meta?["status_code"] as? Int else { return }
+        let results = dictResponse["results"] as? NSDictionary
+        let error = results?["error"]  as? NSDictionary
+        guard let message = error?["message"] else { return }
+        self.error = NSError.init(domain: "Pelias", code: code, userInfo: [NSLocalizedDescriptionKey:message])
+      }
     }
   }
 
-  fileprivate func parseData(_ data: Data?) -> NSDictionary? {
+  fileprivate func parseData<T>(_ data: Data?) -> T?  {
     guard let JSONData = data else { return nil }
     do {
-      let JSON = try JSONSerialization.jsonObject(with: JSONData, options:JSONSerialization.ReadingOptions(rawValue: 0))
-      guard let JSONDictionary :NSDictionary = JSON as? NSDictionary else {
-        print("Not a Dictionary")
-        // put in function
-        return nil
-      }
-      print("JSONDictionary! \(JSONDictionary)")
-      return JSONDictionary
+      return try JSONSerialization.jsonObject(with: JSONData, options:JSONSerialization.ReadingOptions(rawValue: 0)) as? T
```

Comment:
Try `JSONSerialization.ReadingOptions(.mutableContainers)` or something like that. Shouldn't need rawValue anymore.